internal/api/v2/types_gen.go

```diff
@@ -53,6 +53,11 @@ type QueryResponse struct {
 	Status      interface{}  `json:"status,omitempty" form:"status" query:"status" validate:"required"`
 }
 
+type DirectoryQueryResult struct {
+	Entries         []string         `json:"entries,omitempty" form:"entries" query:"entries" validate:"required"`
+	ExpandedEntries []*QueryResponse `json:"expandedEntries,omitempty" form:"expandedEntries" query:"expandedEntries" `
+}
```

Comment:
I also chose for a separate field name. Returning different structures for the same rest call can be painful to unmarshal in some languages.